crates/matrix-sdk-ui/src/room_list/room_list.rs

```diff
@@ -58,3 +106,44 @@ impl RoomList {
         self.sliding_sync_list.room_list_filtered_stream(filter)
     }
 }
+
+/// The loading state of a [`RoomList`].
+///
+/// When a [`RoomList`] is displayed to the user, it can be in various states.
+/// This enum tries to represent those states with a correct level of
+/// abstraction.
+#[derive(Clone, Debug)]
+pub enum RoomListLoadingState {
+    /// The [`RoomList`] has not been loaded yet, i.e. a sync might run
+    /// or not run at all, there is nothing to show in this `RoomList` yet.
+    /// It's a good opportunity to show a placeholder to the user.
+    ///
+    /// From [`Self::NotLoaded`], it's only possible to move to
+    /// [`Self::Loaded`].
+    NotLoaded,
+
+    /// The [`RoomList`] has been loaded, i.e. a sync has been run, or more
+    /// syncs are running, there is probably something to show to the user.
+    /// Either the user has 0 room, in this case, it's a good opportunity to
+    /// show a special screen for that, or the user has multiple rooms, and it's
+    /// the classical room list.
+    ///
+    /// The number of rooms is represented by `maximum_number_of_rooms`.
+    ///
+    /// From [`Self::Loaded`], it's not possible to move back to
+    /// [`Self::NotLoaded`].
+    Loaded {
+        /// The maximum number of rooms a [`RoomList`] contains.
+        ///
+        /// It does not mean that there is exactly this amount of rooms to
+        /// display. Usually, the room entries are represented by
+        /// [`RoomListEntry`]. The room entry might have been synced or not
+        /// synced yet, but we know for sure (from the server), that there will
+        /// be this amount of rooms in the list at the end.
```

Comment:
For a list in growing/paginating mode, this will increase over time, so this comment might be a bit misleading in that case, is that a big deal? In this case, we might either 1. wait for the number of maximum_number_of_rooms to stabilize, and emit the state update after that, or 2. introduce back `PartiallyLoaded` for that in-between state?

Reply:
No, it won't increase. The server replies with the maximum number of rooms even we don't load all the rooms yet. It's the total number, the maximum number, of available rooms we can fetch from this SS list.